databuilder/query_language.py

```diff
@@ -50,11 +53,16 @@ def __setattr__(self, name, value):
             raise TypeError(
                 f"Invalid column '{name}'. Dataset columns must return one row per patient"
             )
-        super().__setattr__(name, value)
+        self.variables[name] = value
```

Comment:
I'm not set on this change. Seems slightly better to be more explicit, but since we're overriding `__setattr__` I don't think that there is a completely clean implementation.

Reply:
I think this is a definite improvement, thanks for doing it.